packages/flutter/lib/src/painting/path_border.dart

```diff
@@ -0,0 +1,191 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'package:flutter/foundation.dart';
+
+import 'border_radius.dart';
+import 'borders.dart';
+import 'box_border.dart';
+import 'edge_insets.dart';
+
+/// The signature of a method that creates a [Path] for a [PathBorder].
+///
+/// The `bounds` parameter will not be null.
+typedef PathBorderBuilder = Path Function(Rect bounds, TextDirection textDirection);
+
+/// A [BoxBorder] that delegates painting its border to a [PathBorderBuilder].
+///
+/// This class can be used draw a customized border as a [BoxDecoration], for
+/// example a path that has been dashed or trimmed.
+///
+/// {@tool sample}
+/// This will draw a container with a red elliptical border:
+///
+/// ```dart
+/// Container(
+///   decoration: BoxDecoration(
+///     border: PathBorder(
+///       pathBuilder: (Rect bounds, TextDirection textDirection) =>
+///           Path()..addOval(bounds),
+///       border: const BorderSide(color: Colors.red),
```

Comment:
every time this gets built it'll create a different border, it'll fail to lerp between them, etc. This seems suboptimal.

Reply:
We don't have a good way to lerp between paths.

In terms of building each time, I considered caching the path and transforming it but that gets much harder to reason about for the builder side - what if you dash your path and now it's getting scalled bigger in one dimension than the other and it's not dashing the way you want anymore?

Reply:
Or do you just mean in the example having it return a path that's already been created instead of drawinga  new one each time?

Reply:
I mean that right now, the way it's designed, the most obvious way to use the API is that every time the widget that uses this shape gets built, it'll think there's a new shape (because it won't be == to the old one), and it'll try to lerp between the old shape and the new shape, which will fail (since this class doesn't know how to lerp with itself), which will result in ugly renderings (I forget what the default behaviour is for lerping shapes but it isn't ideal, I'm sure).